internal/store/index.go

```diff
@@ -93,8 +93,13 @@ func (s *ObjectStore) createIndexes(ctx context.Context,
 				return err
 			}
 
-			err = tx.Put(ctx, key, value)
+			err = tx.Insert(ctx, key, value)
```

Comment:
I'm not sure this is what we want to do here. If an object is being created (not upserted), this is fine, however in the event it's being upserted, wouldn't this always block the upsert as long as there's a unique index on that object?

This code path is called for both create & upsert.

Reply:
The Upsert code is already deleting indexes then recreate them. https://github.com/Kong/koko/blob/main/internal/store/store.go#L193

Reply:
Ohh derp, right.